libafl/src/stages/gen.rs

```diff
@@ -0,0 +1,74 @@
+//! A [`Stage`] that generates a single input via a [`Generator`] and evaluates
+//! it using the fuzzer, possibly adding it to the corpus.
+
+use core::marker::PhantomData;
+
+use crate::{
+    generators::Generator,
+    inputs::UsesInput,
+    stages::Stage,
+    state::{HasCorpus, HasRand, UsesState},
+    Error, Evaluator,
+};
+
+/// A [`Stage`] that generates a single input via a [`Generator`] and evaluates
+/// it using the fuzzer, possibly adding it to the corpus.
+///
+/// This stage can be used to construct black-box (e.g., grammar-based) fuzzers.
+#[derive(Debug)]
+pub struct GenStage<Z, G>(G, PhantomData<Z>)
```

Comment:
Please order these type parameters alphabetically. (so it should be GenStage<G, Z>)